degree-importer/degree-importer.php

```diff
@@ -0,0 +1,595 @@
+                'degree_type_id'         => $program->type_id,
+                'degree_hours'           => $program->required_hours,
+                'degree_description'     => html_entity_decode( $program->description ),
+                'degree_website'         => $program->website,
+                'degree_phone'           => $program->phone,
+                'degree_email'           => $program->email,
+                'degree_contacts'        => $program->contacts,
+                'degree_pdf'             => $program->catalog_url,
+                'degree_is_graduate'     => $program->graduate,
+                'page_header_height' => 'header-media-default'
+            ),
+            'post_terms' => array(
+                'program_types' => $program->type,
+                'colleges'      => $program->college_name,
+                'departments'   => $program->department_name
+            )
+        );
+    }
+
+    /**
+     * Creates or returns the post
+     * @author Jim Barnes
+     * @since 1.0.0
+     * @param $post_data Array | The array of post data
+     * @param $degree_id int | The degree id
+     * @return int | The post id
+     **/
+    private function process_post( $post_data, $degree_id, $degree_type_id=null ) {
+        $retval = null;
+
+        // Attempt to fetch an existing post
+        $args = array(
+            'post_type'      => $post_data['post_type'],
+            'posts_per_page' => 1,
+            'meta_query'     => array(
+                array(
+                    'key'   => 'degree_id',
+                    'value' => $degree_id
+                )
+            ),
+            'tax_query'      => array(
+                array(
+                    'taxonomy' => 'program_types',
+                    'field'    => 'slug',
+                    'terms'    => sanitize_title( $post_terms['program_types'] )
+                )
+            )
+        );
+
+        if ( $degree_type_id ) {
+            $args['meta_query'][] = array(
+                'key'   => 'degree_type_id',
+                'value' => $degree_type_id
+            );
+        }
+
+        $existing_post = get_posts( $args );
+        $existing_post = empty( $existing_post ) ? false : $existing_post[0];
+
+        if ( $existing_post !== false ) {
+            $retval = $existing_post->ID;
+            $post_data['ID'] = $post_id;
+            $post_data['post_status'] = $existing_post->post_status;
+
+			// Remove the post name so we're not updating permalinks
+			unset( $post_data['post_name'] );
+
+            // Remove the post date so publish date stays the same.
+            unset( $post_data['post_date'] );
+
+            wp_update_post( $post_data );
+
+            // Remove the post from the existing array
+            unset( $this->existing_posts[$post_data['ID']] );
+            $this->existing_count++;
+        } else {
+            $retval = wp_insert_post( $post_data );
```

Comment:
New posts need to be inserted with post_status of 'draft', and then updated to 'published' after remaining posts in $this->existing_posts have been deleted from wordpress (to prevent the incrementing `degree-slug-2`, `degree-slug-3` problem we just fixed: https://github.com/UCF/Main-Site-Theme/compare/v2.3.11...v2.3.12)